File: solutions/0058_length-of-last-word.java
// Problem: https://leetcode.com/problems/length-of-last-word
// Runtime: 0 ms

class Solution {
    public int lengthOfLastWord(String s) {
        int out = 0;
        for (int i = s.length() - 1; i >= 0; i--) {
            if (s.charAt(i) == ' ') {
                if (out == 0) {
                    continue;
                }
                return out;
            }
            out++;
        }
        return out;
    }
}

File: solutions/0058_length-of-last-word.py
# Problem: https://leetcode.com/problems/length-of-last-word
# Runtime: 36 ms

class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        curr = 0
        firstC = False
        for c in s[::-1]:
            if c == ' ':
                if firstC:
                    return curr
            else:
                curr += 1
                firstC = True
        return curr